### PR TITLE
Set `StrictMode` version `3.0`

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

`Set-StrictMode -Version 3.0`

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by xtqqczze** [comment]
CodeFactor issues are preexisting.

File: build.psm1 [hunk 1]
```diff
@@ -837,7 +843,7 @@ function Get-PSOptions {
         $DefaultToNew
     )
 
-    if(!$script:Options -and $DefaultToNew.IsPresent)
+    if ($null -eq $script:Options -and $DefaultToNew.IsPresent)
```
Comment (TravisEz13) [inline build.psm1]:
```suggestion
    if (!$script:Options -and $DefaultToNew.IsPresent)
```
Please use standard syntax for this

File: build.psm1 [hunk 2]
```diff
@@ -860,7 +866,7 @@ function Get-PSOutput {
     )
     if ($Options) {
         return $Options.Output
-    } elseif ($script:Options) {
+    } elseif ($null -ne $script:Options) {
```
Comment (TravisEz13) [inline build.psm1]:
```suggestion
    } elseif ($script:Options) {
```
Please use standard syntax

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments

**Comment by xtqqczze** [comment]
CodeFactor issues are preexisting.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@xtqqczze Thanks for the contribution

**Pull request merged**